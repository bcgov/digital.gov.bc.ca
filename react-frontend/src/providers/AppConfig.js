import React, { Component } from 'react';

export const AppConfigContext = React.createContext();

export class AppConfig extends Component {
  // TODO: It may be best to move the config set state into
  // constructor to make sure the config is assigned
  constructor() {
    super();
    this.state = { strapiApiUrl: null };
  }

  componentDidMount() {
    const xhr = new XMLHttpRequest();
    xhr.open('GET', '/config.json');

    xhr.send();

    xhr.onload = () => {
      if (xhr.status !== 200) {
        // analyze HTTP status of the xhrs response
        // do something
      } else {
        this.setState(JSON.parse(xhr.response));
      }
    };
  }

  render() {
    return (
      <AppConfigContext.Provider value={{ state: this.state }}>
        {this.props.children}
      </AppConfigContext.Provider>
    );
  }
}
