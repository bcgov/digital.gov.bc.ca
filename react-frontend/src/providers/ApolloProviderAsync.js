import React, { useContext, useEffect, useState } from 'react';
import { ApolloProvider } from 'react-apollo';
import ApolloClient from 'apollo-boost';

import { AppConfigContext } from './AppConfig';

// The apollo provider implements the strapi configuration
// by pulling the backend url from a config file. This allows
// the deployment config file instead of being baked into
// the build config.

function ApolloProviderAsync(props) {
  const config = useContext(AppConfigContext);
  const strapiURL = config['state']['strapiApiUrl'];

  // using client as state prevents it from being initialized needlessly
  const [client, setClient] = useState(
    new ApolloClient({
      uri: 'http://localhost:3000/graphql',
    })
  );

  // The useEffect is only called when config changes
  useEffect(() => {
    if (strapiURL) {
      setClient(new ApolloClient({ uri: strapiURL }));
    }
  }, [config, strapiURL]);

  return <ApolloProvider client={client}>{props.children}</ApolloProvider>;
}

export default ApolloProviderAsync;
