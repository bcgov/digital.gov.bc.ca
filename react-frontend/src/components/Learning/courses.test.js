import React from 'react';
import renderer from 'react-test-renderer';
import Courses from './courses';
import { MockedProvider } from '@apollo/react-testing';
import { AppConfig } from '../../providers/AppConfig';

it('renders snapshot correctly', () => {
  const tree = renderer
    .create(
      <AppConfig>
        <MockedProvider>
          <Courses />
        </MockedProvider>
      </AppConfig>
    )
    .toJSON();
  expect(tree).toMatchSnapshot();
});
