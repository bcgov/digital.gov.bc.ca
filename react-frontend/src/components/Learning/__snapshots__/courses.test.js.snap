// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders snapshot correctly 1`] = `
<div
  className="sc-hKwCoD sc-gKckTs fBQKBj kFOath horizontalAlignment contentBlockContainer container"
  id="courses"
>
  <div
    className="row"
  >
    <div
      className="col-sm-12"
    >
      <h2
        className="sc-bqiQRQ bGyPlm heading"
      >
        Courses
      </h2>
    </div>
  </div>
  <p>
    Loading...
  </p>
</div>
`;
